src/NServiceBus.Core/Performance/Statistics/ReceivePerformanceDiagnosticsBehavior.cs

```diff
@@ -9,12 +9,22 @@ namespace NServiceBus
     [SkipWeaving]
     class ReceivePerformanceDiagnosticsBehavior : Behavior<PhysicalMessageProcessingContext>
     {
-   
+        public ReceivePerformanceDiagnosticsBehavior(string transportAddress)
```

Comment:
Isn't this change unnecessary including the above feature setup? Since `PhysicalMessageProcessingContext` inherits from `IncomingContext` you have access to the pipeline info.